ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1760,6 +1765,79 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     return Status;
 }
 
+static
+ULONG
+IopValidateID(IN PWCHAR Id,
+              IN SIZE_T MaxIdLen,
+              IN ULONG MaxSeparators,
+              IN BOOLEAN IsMultiSz)
+{
+   PWCHAR PtrSymbol;
+   PWCHAR StringEnd;
+   WCHAR Symbol;
+   ULONG SeparatorsCount = 0;
+   PWCHAR PtrPrevSymbol = NULL;
+
+   PAGED_CODE();
+
+   StringEnd = Id + MaxIdLen;
+
+   for (PtrSymbol = Id; PtrSymbol < StringEnd; PtrSymbol++)
+   {
+      Symbol = *PtrSymbol;
+
```

Comment:
You may also check for the different allowed forms, for example (for Hardware Ids) those mentioned in https://docs.microsoft.com/en-us/windows-hardware/drivers/install/hardware-ids ...

Reply:
I check in real conditions: these ISOs ( http://vgal.ru.com/reactos-0-4-7-2-new-usb/ ) have already applied this patch.

I did this patch because the usbhub_new driver did not have a final UNICODE_NULL. Because of this, BSOD 0x50 sometimes appeared.